docs/references/architecture/adr-x-intelligent-validator-peering.md

```diff
@@ -0,0 +1,78 @@
+
+First, we modify the handshake that nodes perform when connecting to each other. Currently, that handshake exchanges `NodeInfo`. We will augment `NodeInfo` to include an `isValidator` field, which is just the `isValidator` value from the config. This will be kept for each peer of the node in the peer's `NodeInfo`.
+
+We will add a new reactor for validator peering, which we will refer to here as valp2p. All nodes in the network should run this new reactor.
+
+The valp2p reactor overrides the `AddPeer` and `Receive` reactor methods.
+
+`AddPeer`: this is called by the switch on every reactor when a peer is added. For valp2p, this should:
+1. Check if the peer is validator, if it is not, return.
+2. If the peer is validator, record its net address in memory. Then, broadcast its net address to all peers.
+
+`Receive`: this processes the messages received by this reactor.
+3. Parse and validate the message. The message should contain a valid net address.
+4. If the current node is a full node, broadcast the message to all peers.
+5. If the current node is a validator, record the net address in memory, then broadcast the message to all peers.
+
+Additionally, when this reactor starts, it should start a goroutine that periodically:
+1. Checks the number of validator peers we are currently connected to.
+2. If the number < `val_peer_count_low`, start dialing random validator addresses that we know about until we reach `val_peer_count_target`.
+3.  If the number > `val_peer_count_high`, drop some random validators until we reach `val_peer_count_target`.
+
+[ The exact logic of this goroutine may need to be tweaked, it may interfere with persistent peers, maxnuminboundpeers, and maxnumoutboundpeers. ]
+
+The `AddPeer` behavior is roughly: when I get connected to a new validator, tell my neighbors about this new validator so they can potentially connect as well, if they are a validator.
+
+The `Receive` behavior is roughly: when I hear about a new validator, keep it in my validator address book so I can potentially connect to it later, and let my neighbors know about it so they can do the same.
+
+[ May need to add some `broadcastHasValidatorAddress` type of thing to tell other nodes not to send me a specific validator address, similar to consensus `broadcastHasVote` ]
+
+This should probabilistically achieve a connected subnetwork of validators with a small diameter, although additional thought needs to be put into how to set the val peer bounds and target based on the number of validators in the network.
+
+
+## Consequences
+
+> This section describes the consequences, after applying the decision. All
+> consequences should be summarized here, not just the "positive" ones.
+
+### Positive
+- The validator subnetwork diameter will be smaller, allowing messages to propagate between validators with less latency
+
+### Negative
+- There is additional network bandwidth used by the new reactor
```

Comment:
I think the additional network bandwidth is going to be pretty negligible here considering a lot of the additional data is on handshake, or via infrequent heartbeat-like pex behavior. 

Separately, we can explore the addition of options to disable gossip that is not relevant to different node types. For example, someone running a non-validating node might not care about getting tx mempool gossip.